plugins/org.python.pydev/tests_analysis/com/python/pydev/analysis/refactoring/quick_fixes/AddTokenAndImportStatementTest.java

```diff
@@ -410,6 +410,104 @@ public void testAllCases() throws Exception {
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult\r\n" +
+                "";
+        String expectedDoc = "from mod2 import \\\r\n" +
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict\r\n" +
+                "";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
+
+    public void testGroupImport4() throws Exception {
+        String baseDoc = "from mod2 import \\\r\n" +
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult,\\";
+        String expectedDoc = "from mod2 import \\\r\n" +
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict,\\";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
+
+    public void testGroupImport5() throws Exception {
+        String baseDoc = "from mod2 import \\\r\n" +
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult,\\\r\n" +
+                "";
+        String expectedDoc = "from mod2 import \\\r\n" +
+                "    ActivityInfoDict,\\\r\n" +
+                "    WorkspaceInfoDict,\\\r\n" +
+                "    PackageInfoDict,\\\r\n" +
+                "    ActionResultDict,\\\r\n" +
+                "    UploadActivityParamsDict,\\\r\n" +
+                "    CloudLoginParamsDict,\\\r\n" +
+                "    ActionResult, UploadNewActivityParamsDict,\\\r\n" +
+                "";
+        checkLocalImport(baseDoc, "from mod2 import UploadNewActivityParamsDict", expectedDoc, true);
+    }
```

Comment:
Please test with the following case:

baseDoc:
```
from mod2 import ActivityInfoDict  # comment \
```

import to add:

```
from mod2 import UploadNewActivityParamsDict
```

In this case it should put the `UploadNewActivityParamsDict` in the same line (before the comment). It seems that right now it puts a `, UploadNewActivityParamsDict` in a new line (when the comment ends with the `\` in the previous line).